src/app/Main/CheckServerConnection.component.tsx

```diff
@@ -25,7 +26,7 @@ export function CheckServerConnectionComponent(): JSX.Element {
     });
 
     if (loading) {
-        return <p> </p>;
+        return <LoadingProcessComponent />;

```

Comment:
Question: How does the CheckServerConnectionComponent work? Where it is used and when the connection is checked? And why the connection is checked?

Reply:
This check was written as part of another ticket. It checks if there is in general a connection to the server (/v1). If not, it is not possible to access the query page at all (tab is blocked).

Reply:
Hm, ok. As stated in another comment, i think it would be better to show an error message if the connection gets lost. The initial server connection is already checked by the former 'ChooseComponent'.
Another question: When is this checked? On page load, user interaction or in a time intervall?

Reply:
It was Dominic's idea to do it that way. I have to discuss it with him - if I want to do it differently. The check is done once on page load.

Reply:
Ok, discussion is always good :). So if it is done on page load it seems obsolete. If you discuss this with dominic think of the type of data that you use in your app. What data is essential to get the app working and is loaded at page load and what data is fetched by user interaction. Best would be if there is only one point in the app handling the essential data loading. And as stated, the data fetching should be handled on the fly by displaying error messages (in my opinion)